src/components/products/CustomTab.jsx

```diff
@@ -36,18 +36,28 @@ const buttonConfig = {
 
 const CustomTab = () => {
   const dispatch = useDispatch();
-  const navigate = useNavigate();
   const categories = useSelector(categorySelector);
   const loading = useSelector(loadingSelector);
   const error = useSelector(errorselector);
 
+  const [searchParams, setSearchParams] = useSearchParams();
+
   useEffect(() => {
     dispatch(fetchCategoryData());
-  }, []);
+  }, [dispatch]);
+
+  const handleCategorySelect = (categorySlug) => {
+    dispatch(setActiveCategory(categorySlug));
```

Comment:
remove this 
activeCategory is not getting used anywhere 
remove the set dispatch related to activeCategory from everywhere

Reply:
Done.